src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,951 @@
+
+            // Need to special case for expressions that are contained within a parameter
+            // because it is technically "contained" within a method, but an expression in a parameter does not make
+            // sense to introduce.
+            var expressionInParameter = expression.FirstAncestorOrSelf<SyntaxNode>(node => syntaxFacts.IsParameter(node));
+            if (expressionInParameter is not null)
+            {
+                return;
+            }
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var containingMethod = expression.FirstAncestorOrSelf<SyntaxNode>(node => generator.GetParameterListNode(node) is not null);
+
+            if (containingMethod is null)
+            {
+                return;
+            }
+
+            var containingSymbol = semanticModel.GetDeclaredSymbol(containingMethod, cancellationToken);
+            if (containingSymbol is not IMethodSymbol methodSymbol)
+            {
+                return;
+            }
+
+            var methodKind = methodSymbol.MethodKind;
+            if (methodKind is not (MethodKind.Ordinary or MethodKind.LocalFunction or MethodKind.Constructor))
+            {
+                return;
+            }
+
+            var actions = await GetActionsAsync(document, expression, methodSymbol, containingMethod, cancellationToken).ConfigureAwait(false);
+
+            if (actions is null)
+            {
+                return;
+            }
+
+            context.RegisterRefactoring(new CodeActionWithNestedActions(
+                CreateDisplayText(expression, syntaxFacts, FeaturesResources.Introduce_parameter_for_0), actions.Value.actions, isInlinable: false), textSpan);
+            context.RegisterRefactoring(new CodeActionWithNestedActions(
+                CreateDisplayText(expression, syntaxFacts, FeaturesResources.Introduce_parameter_for_all_occurrences_of_0), actions.Value.actionsAllOccurrences, isInlinable: false), textSpan);
+        }
+
+        /// <summary>
+        /// Introduces a new parameter and refactors all the call sites.
+        /// </summary>
+        public async Task<Solution> IntroduceParameterAsync(Document document, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, SyntaxNode containingMethod, bool allOccurrences, bool trampoline, bool overload,
+            CancellationToken cancellationToken)
+        {
+            var parameterName = await GetNewParameterNameAsync(document, expression, cancellationToken).ConfigureAwait(false);
+
+            var methodCallSites = await FindCallSitesAsync(document, methodSymbol, cancellationToken).ConfigureAwait(false);
+
+            return await RewriteSolutionAsync(document,
+                expression, methodSymbol, containingMethod, allOccurrences, parameterName, methodCallSites,
+                trampoline, overload, cancellationToken).ConfigureAwait(false);
```

Comment:
nit.  consider inlining this method.  being a different method doesn't buy you much.  up to you though.